src/components/atoms/Scrollbar/Scrollbar.tsx

```diff
@@ -0,0 +1,201 @@
+    /**
+     * Automatically scrolls the container to a specific vertical position (in pixels).
+     */
+    autoScrollTopTo?: number;
+    /**
+     * Automatically scrolls the container to a specific horizontal position (in pixels).
+     */
+    autoScrollLeftTo?: number;
+    /**
+     * Callback function triggered during scroll events. Receives the scroll event as an argument.
+     */
+    onScroll?: (e: UIEvent<HTMLSpanElement>) => void;
+    /**
+     * 	Custom height for the scrollable container.<br>
+     * 	Default is 100%.<br>
+     * 	Possible values: `any valid css value`
+     */
+    customHeight?: string;
+    /**
+     * 	Custom width for the scrollable container.<br>
+     * 	Default is 100%.<br>
+     * 	Possible values: `any valid css value`
+     */
+    customWidth?: string;
+}
+
+/**
+ * Scrollbar is a UI element that allows users to navigate through content that extends beyond the visible area of a container or window. It typically appears along the right side or bottom of the viewport, providing a draggable handle and directional arrows for vertical or horizontal scrolling, enabling users to access all available content.
+ */
+const Scrollbar: FC<IScrollbarProps> = ({
+    className,
+    children,
+    autoScrollTopTo,
+    autoScrollLeftTo,
+    onScroll,
+    customHeight,
+    customWidth
+}) => {
+    const [scrollDirection, setScrollDirection] = useState<"x" | "y" | null>(null);
+    const previousScrollPosition = useRef({ scrollTop: 0, scrollLeft: 0 });
+    const scrollbarsRef = useRef<(ScrollBarTypes & HTMLDivElement) | null>(null);
+
+    const scrollStateResetHandler = () => {
+        setScrollDirection(null);
+    };
+
+    const { debouncedCallback, clearDebounce } = useDebouncedCallback(scrollStateResetHandler, 1000);
+
+    const scrollHandler = (e: UIEvent<HTMLSpanElement>) => {
+        onScroll?.(e);
+        const target = e.target as HTMLSpanElement;
+
+        const { scrollTop, scrollLeft } = target;
+
+        const previous = previousScrollPosition.current;
+
+        const deltaY = scrollTop - previous.scrollTop;
+        const deltaX = scrollLeft - previous.scrollLeft;
+
+        if (Math.abs(deltaY) > Math.abs(deltaX)) {
+            setScrollDirection("y");
+        } else if (Math.abs(deltaX) > Math.abs(deltaY)) {
+            setScrollDirection("x");
+        }
+
+        previousScrollPosition.current = { scrollTop, scrollLeft };
+
+        debouncedCallback();
+    };
+
+    const showScrollbarHandler = (direction: "x" | "y") => {
+        if (scrollDirection === direction) {
+            setScrollDirection(direction);
+            clearDebounce();
+        }
+    };
+
+    const hideScrollbarHandler = () => {
+        debouncedCallback();
+    };
+
+    // autoScrollTo Top and Left
+    useEffect(() => {
+        const scrollRefCurrent = scrollbarsRef.current;
+        if (scrollRefCurrent?.contentElement?.parentElement && (autoScrollTopTo || autoScrollLeftTo)) {
```

Comment:
Here is the condition will not work if I will use the value `0` for the `autoScrollTopTo` or `autoScrollLeftTo` and try to change checking something like `!== null` or `undefined`